### ISSUE TITLE
Stop showing trips that are impossible to take

### ISSUE THREAD

**Issue by M00NJ** [grote/Transportr#940]
**Describe the bug**
When planing a Trip in the future it makes sense to show multiple options before and after the selected time of arrival, so the user can pick the best one. However it makes no sense to show options from the past that are impossible to take. None at all or maybe a maximum of 5min ago, in case the user wants to run I guess. The current situation is not optimal though. The whole screen is filled with trips that can't be taken as seen in the screenshot. It just doesn't make sense to fill up the ui with useless information like that.

**To Reproduce**
1. Select a location and destination
2. Select a time of arrival today, that's still possible to arrive at, but not too far off in the future
3. Transportr shows trips from the past

**Expected behavior**
Transportr doesn't show trips that are impossible to take

**Screenshots**
![signal-2024-03-31-13-13-22-592](https://github.com/grote/Transportr/assets/82398591/7325ba2e-f857-4d18-a399-c4b058f80010)

**Versions (please complete the following information):**
 - **Transportr Version:** 2.2.1
 - **Device:** Pixel 4a 5G
 - **Android Version:** 14 (GrapheneOS)

**Comment by JoKeyser** [comment]
I disagree. To me, including trips in the past are a feature, not a bug.

There are at least two use cases:

1. I'm already taking a connection from A to B and want to see that ongoing connection (with start in the past), to figure out what to do next.
2. I want to pick up somebody from the station and know their starting time (in the past) but would like to check when and where they are arriving.

**Comment by M00NJ** [comment]
Valid points. The way most other applications solve this is by having a little button that says `earlier`. That way the screen doesn't get filled a with trips that are useless most of the time, but if these trips are need like in the situations you described, they can still be displayed.

**Comment by JoKeyser** [comment]
Nice, such an option to easily display earlier trips on demand would be OK for me.
